src/Analysis/Codelyzer.Analysis.Build/WorkspaceBuilderHelper.cs

```diff
@@ -459,6 +461,40 @@ private EnvironmentOptions GetEnvironmentOptions(IProjectFile projectFile)
                 }
             }
 
+            try
+            {
+                MSBuildLocator.RegisterDefaults();
+            }
+            catch (System.InvalidOperationException)
+            {
+                var enviroment = System.Environment.CurrentDirectory;
+                string projectDirectory = Directory.GetParent(enviroment).Parent.FullName;
+                var proc = new Process
+                {
+                    StartInfo = new ProcessStartInfo
+                    {
+                        FileName = Path.Combine(projectDirectory, "vswhere.exe"),
+                        Arguments = "-latest -requires Microsoft.Component.MSBuild -find MSBuild\\**\\Bin\\MSBuild.exe",
+                        UseShellExecute = false,
+                        RedirectStandardOutput = true,
+                        CreateNoWindow = true,
+                        WorkingDirectory = ""
+                    }
+                };
+                proc.Start();
+                var msbuildExe = "";
+                while (!proc.StandardOutput.EndOfStream)
+                {
+                    msbuildExe = proc.StandardOutput.ReadLine();
```

Comment:
Are we always sure it's a one line output?

Reply:
Yes, it should always give the lastest vs installation in one line or nothing at all which will be covered in the next catch block.